tests/llms/test_cohere.py

```diff
@@ -0,0 +1,149 @@
+            "prompt": "test prompt",
+            "meta": {"api_version": {"version": "1"}},
+        },
+        return_likelihoods=False,
+    )
+
+
+async def mock_acompletion_with_retry(*args: Any, **kwargs: Any) -> dict:
+    # Example taken from https://docs.cohere.com/reference/generate
+    return cohere.responses.Generations.from_dict(
+        {
+            "id": "21caa4c4-6b88-45f7-b144-14ef4985384c",
+            "generations": [
+                {
+                    "id": "b5e2bb70-bc9c-4f86-a22e-5b5fd13a3482",
+                    "text": "\n\nThis is indeed a test",
+                    "finish_reason": "COMPLETE",
+                }
+            ],
+            "prompt": "test prompt",
+            "meta": {"api_version": {"version": "1"}},
+        },
+        return_likelihoods=False,
+    )
+
+
+def mock_chat_with_retry(*args: Any, **kwargs: Any) -> dict:
+    return cohere.responses.Chat.from_dict(
+        {
+            "chatlog": None,
+            "citations": None,
+            "conversation_id": None,
+            "documents": None,
+            "generation_id": "357d15b3-9bd4-4170-9439-2e4cef2242c8",
+            "id": "25c3632f-2d2a-4e15-acbd-804b976d0568",
+            "is_search_required": None,
+            "message": "test prompt",
+            "meta": {"api_version": {"version": "1"}},
+            "preamble": None,
+            "prompt": None,
+            "response_id": "25c3632f-2d2a-4e15-acbd-804b976d0568",
+            "search_queries": None,
+            "search_results": None,
+            "text": "\n\nThis is indeed a test",
+            "token_count": {
+                "billed_tokens": 66,
+                "prompt_tokens": 64,
+                "response_tokens": 9,
+                "total_tokens": 73,
+            },
+        },
+        client=None,
+        message="test_prompt",
+    )
+
+
+async def mock_achat_with_retry(*args: Any, **kwargs: Any) -> dict:
+    return cohere.responses.Chat.from_dict(
+        {
+            "chatlog": None,
+            "citations": None,
+            "conversation_id": None,
+            "documents": None,
+            "generation_id": "357d15b3-9bd4-4170-9439-2e4cef2242c8",
+            "id": "25c3632f-2d2a-4e15-acbd-804b976d0568",
+            "is_search_required": None,
+            "message": "test prompt",
+            "meta": {"api_version": {"version": "1"}},
+            "preamble": None,
+            "prompt": None,
+            "response_id": "25c3632f-2d2a-4e15-acbd-804b976d0568",
+            "search_queries": None,
+            "search_results": None,
+            "text": "\n\nThis is indeed a test",
+            "token_count": {
+                "billed_tokens": 66,
+                "prompt_tokens": 64,
+                "response_tokens": 9,
+                "total_tokens": 73,
+            },
+        },
+        client=None,
+        message="test_prompt",
+    )
+
+
+@pytest.mark.skipif(cohere is None, reason="cohere not installed")
```

Comment:
If you wanted, since you're already patching in the `mock_completion_with_retry`, you could also patch in the `cohere` library, and remove these `skipif`s.

I know this `skipif` is a practice other LLM tests use, but they're not ideal because what happens is these tests don't get run.

Reply:
Do you mean create a mock cohere object? That would require me to also create a mock cohere.Generations, cohere.Chat I can do that, I just want to make sure I understand what you mean.

Reply:
Oh I see.  I think if a `cohere` object is a `MagicMock`, it will automatically have `cohere.Client` and `cohere.Client.chat` as sub-mocks.  Though does it matter, because you're patching `completion_with_retry`, which I believe is above the `client` level.

Also, please feel free to ignore this comment on patching, was just sharing.

Reply:
the return type from completion_with_retry is the problem. Right now I'm using cohere.responses.Generations.from_dict, and cohere.responses.Chat.from_dict to generate what would be a realistic return value. I could mock the return type to just return what is required, which would be obj.text and obj.generations[0].text where obj is the return value but that wouldn't give someone looking at the test and idea of what is returned by the completion_with_retry function